tracking/R3BMDFWrapper.cxx

```diff
@@ -0,0 +1,215 @@
+/******************************************************************************
+ *   Copyright (C) 2019 GSI Helmholtzzentrum für Schwerionenforschung GmbH    *
+ *   Copyright (C) 2019 Members of R3B Collaboration                          *
+ *                                                                            *
+ *             This software is distributed under the terms of the            *
+ *                 GNU General Public Licence (GPL) version 3,                *
+ *                    copied verbatim in the file "LICENSE".                  *
+ *                                                                            *
+ * In applying this license GSI does not waive the privileges and immunities  *
+ * granted to it by virtue of its status as an Intergovernmental Organization *
+ * or submit itself to any jurisdiction.                                      *
+ ******************************************************************************/
+#include"R3BMDFWrapper.h"
+#include <cmath>
+#include<iostream>
+#include<iomanip>
+#include<fstream>
+
+R3BMDFWrapper::R3BMDFWrapper()
+    : mdf_NVariables(0)
+    , pca_NVariables(0)
+    , is_PCA(false)
+{
+}
+
+R3BMDFWrapper::R3BMDFWrapper(const char* mdf_file)
+{
+    InitMDF(mdf_file);
+    is_PCA=false;
+}
+
+R3BMDFWrapper::~R3BMDFWrapper()
+{
+}
+
+//This method will read a special txt file with MDF parameters, don't change
+void R3BMDFWrapper::InitMDF(const char* mdf_file)
+{
+    int i;
+    std::ifstream file(mdf_file);
+    if(file.is_open())
+    {
+        file >> mdf_NVariables >> mdf_NCoefficients >> mdf_NMaxFunctions
+            >> mdf_DMean >> mdf_PolyType;
+
+        
+        mdf_XMean = new double[mdf_NVariables];
+        for(i=0; i<mdf_NVariables; i++)
+            file >> mdf_XMean[i];
+
+        mdf_XMin = new double[mdf_NVariables];
+        for(i=0; i<mdf_NVariables; i++)
+            file >> mdf_XMin[i];
+
+        mdf_XMax = new double[mdf_NVariables];
+        for(i=0; i<mdf_NVariables; i++)
+            file >> mdf_XMax[i];
+
+        mdf_Coefficient = new double[mdf_NCoefficients];
+        for(i=0; i<mdf_NCoefficients; i++)
+            file >> mdf_Coefficient[i];
+
+        mdf_CoefficientRMS = new double[mdf_NCoefficients];
+        for(i=0; i<mdf_NCoefficients; i++)
+            file >> mdf_CoefficientRMS[i];
+
+        mdf_Power = new int[mdf_NCoefficients*mdf_NVariables];
+        for(i=0; i<mdf_NCoefficients*mdf_NVariables; i++)
+            file >> mdf_Power[i];
+        file.close();
+    }
+
+    else 
+        std::cout << "\nUnable to open MDF parameter file!!";
+    return;
+
+}
+
+void R3BMDFWrapper::PrintMDF()
+{
+    std::cout << "\n\n======== Initialized MDF from the R3BMDFWrapper:";
+
+    std::cout << std::setprecision(9) << "\nmdf_NVariables = "  << mdf_NVariables
```

Comment:
It does not use the FAIRRoot/R3BRoot LOG standard which would be preferible for comments (it includes function and line where the call is made).

Reply:
thanks, I will fix all messages to use the usual LOG standard